pylintrc

```diff
@@ -23,18 +30,26 @@
 # R1702: too-many-nested-blocks
 # R1705: no-else-return
 # R1710: inconsistent-return-statements
-# W0110: deprecated-lambda
-# W0141: bad-builtin
+# R1714: Consider merging these comparisons with 'in'
+# R1721: Unnecessary use of a comprehension
+# R1732: Consider using 'with' for resource-allocating operations
+# R1734: Consider using [] instead of list()
+# R1735: use-dict-literal
+# W0105: String statement has no effect
 # W0201: attribute-defined-outside-init
 # W0212: protected-access
-# W0401: wildcard-import
+# W0246: Useless parent or super() delegation
 # W0603: global-statement
+# W0612: Unused variable
 # W0613: unused-argument
 # W0621: redefined-outer-name
-# W0703: broad-except
+# W0707: Consider explicitly re-raising
+# W0718: Catching too general exception Exception
 # W1201: logging-not-lazy
-# W1505: deprecated-method
-disable=C0111,C0201,C0301,C0325,C1801,F0401,W0511,W0142,W0622,C0103,E1101,E1111,R0902,R0912,R0913,R0914,R0915,R1702,R1705,R1710,W0110,W0141,W0201,W0401,W0603,W0212,W0613,W0621,W0703,W1201,W1505
+# W1203: Use lazy % or % formatting in logging functions
+# W1404: Implicit string concatenation found in call
+# W1514: Using open without explicitly specifying an encoding
+disable=C0103,C0114,C0115,C0116,C0201,C0206,C0209,C0301,C0413,C0415,C1802,F0401,W0511,W0622,E0611,E1101,R0801,R0902,R0912,R0913,R0914,R0915,R1702,R1705,R1710,R1714,R1721,R1732,R1734,R1735,W0105,W0201,W0212,W0246,W0603,W0612,W0613,W0621,W0707,W0718,W1201,W1203,W1404,W1514
```

Comment:
What's the reasoning to disable many more warnings?

Reply:
There were a bunch already disabled and flagged as 'todo'.  However some were causing errors when running pylint as they were deprecated or non-existent.
So instead I went through and worked out all the ones currently causing warnings and disabled, and added to the todo list above that line.

Reply:
This offers two advantages:
 - the todo list can now accurately be worked through
 - it's possible to include pylint as part of the CI to ensure no additional pylint warnings happen, and the existing ones can slowly be isolated and removed/fixed.